filters/include/pcl/filters/experimental/grid_filter_base.h

```diff
@@ -0,0 +1,214 @@
+   * voxel to be used
+   */
+  inline void
+  setMinimumPointsNumberPerVoxel(const std::size_t min_points_per_voxel)
+  {
+    min_points_per_voxel_ = min_points_per_voxel;
+  }
+
+  /** \brief Return the minimum number of points required for a voxel to be used.
+   */
+  inline std::size_t
+  getMinimumPointsNumberPerVoxel() const
+  {
+    return min_points_per_voxel_;
+  }
+
+  /** \brief Set to true if leaf layout information needs to be saved for later access.
+   * \param[in] save_leaf_layout the new value (true/false)
+   */
+  inline void
+  setSaveLeafLayout(const bool save_leaf_layout)
+  {
+    save_leaf_layout_ = save_leaf_layout;
+  }
+
+  /** \brief Returns true if leaf layout information will to be saved for later access.
+   */
+  inline bool
+  getSaveLeafLayout() const
+  {
+    return save_leaf_layout_;
+  }
+
+  /** \brief Provide the name of the field to be used for filtering data. In conjunction
+   * with \a setFilterLimits, points having values outside this interval will be
+   * discarded.
+   * \param[in] field_name the name of the field that contains values used for filtering
+   */
+  inline void
+  setFilterFieldName(const std::string& field_name)
+  {
+    filter_field_name_ = field_name;
+  }
+
+  /** \brief Get the name of the field used for filtering. */
+  inline std::string const
+  getFilterFieldName() const
+  {
+    return filter_field_name_;
+  }
+
+  /** \brief Set the field filter limits. All points having field values outside this
+   * interval will be discarded.
+   * \param[in] limit_min the minimum allowed field value
+   * \param[in] limit_max the maximum allowed field value
+   */
+  inline void
+  setFilterLimits(const double& limit_min, const double& limit_max)
+  {
+    filter_limit_min_ = limit_min;
+    filter_limit_max_ = limit_max;
+  }
+
+  /** \brief Get the field filter limits (min/max) set by the user. The default values
+   * are numeric_limits<double>::min(), numeric_limits<double>::max().
+   * \param[out] limit_min the minimum allowed field value
+   * \param[out] limit_max the maximum allowed field value
+   */
+  inline void
+  getFilterLimits(double& limit_min, double& limit_max) const
+  {
+    limit_min = filter_limit_min_;
+    limit_max = filter_limit_max_;
+  }
+
+  /** \brief Set to true if we want to return the data outside the interval specified by
+   * setFilterLimits (min, max).
+   * Default: false.
+   * \param[in] limit_negative return data inside the interval (false) or outside (true)
+   */
+  inline void
+  setFilterLimitsNegative(const bool limit_negative)
+  {
+    filter_limit_negative_ = limit_negative;
+  }
+
+  /** \brief Get whether the data outside the interval (min/max) is to be returned
+   * (true) or inside (false).
+   * \param[out] limit_negative true if data \b outside the interval [min; max] is to be
+   * returned, false otherwise
+   */
+  inline void
+  getFilterLimitsNegative(bool& limit_negative) const
```

Comment:
get? and a boolean by reference?

Reply:
This is copied from the legacy VoxelGrid. IMO it's super weird to have a getter with bool reference. Should we remove this from the API? There are already a getter returning bool

https://github.com/tin1254/pcl/blob/6fc72bd30dd90480ff8a0c580a347e4bab174edf/filters/include/pcl/filters/experimental/grid_filter_base.h#L181

Reply:
This is copied from the legacy VoxelGrid. IMO it's super weird to have a getter with bool reference. Should we remove this from the API? There are already a getter returning bool

https://github.com/tin1254/pcl/blob/6fc72bd30dd90480ff8a0c580a347e4bab174edf/filters/include/pcl/filters/experimental/grid_filter_base.h#L181-L185

Reply:
@mvieth Let's deprecate the original? PCL uses get with our arguments only when the stuff that's copied is a vector or similar (with memory), and uses get with return value otherwise

Reply:
You mean deprecate [this one](https://github.com/PointCloudLibrary/pcl/blob/master/filters/include/pcl/filters/voxel_grid.h#L439) and [this one](https://github.com/PointCloudLibrary/pcl/blob/master/filters/include/pcl/filters/voxel_grid.h#L787)? Yes, agreed. I would suggest to do that in a separate pull request

Reply:
We agreed on only having the getter returning `bool`, I will open a PR later